### ISSUE TITLE
Add support for app_service_plan_id in Windows/Linux slot resources

### ISSUE THREAD

**Issue by NikRimington** [hashicorp/terraform-provider-azurerm#17967]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

The `azurerm_windows_web_app_slot` and `azurerm_linux_web_app_slot` are designed to replace the `azurerm_app_service_slot` however they both appear to be missing a useful bit of functionality. They do not allow for slots to be created against different service plans than the main web app.

The legacy resource had the following option `app_service_plan_id` which meant that your slot's app service plan could differ from the main web apps. It would be great to see this added back in.

### New or Affected Resource(s)/Data Source(s)

azurerm_windows_web_app_slot, azurerm_linux_web_app_slot

### Potential Terraform Configuration

```hcl
resource "azurerm_windows_web_app_slot" "example" {
  name            = "example-slot"
  app_service_id  = azurerm_windows_web_app.example.id
  service_plan_id = azurerm_app_service_plan.alternate-plan.id

  site_config {}
}
```

**Comment by xiaxyi** [comment]
Hey @NikRimington,  the app service plan is included in the app service. the slot can only be created under the same service plan as its parent app service. May I have more backgrounds of this feature request? Thanks!

**Comment by NikRimington** [comment]
Hey @xiaxyi thanks for that feedback. 

Previously with the old `azurerm_app_service_slot` you were able to create slots under a web app against a different app service plan. This is a feature we use as we have all of our dev slots sitting on a different plan (I have no control over that, it's above my paygrade). 

For example, using the `azurerm_app_service` task the following app service was created:
![image](https://user-images.githubusercontent.com/13313745/185196083-41c2a46c-7767-4787-8fe0-e8be22a10be2.png)

Then using the `azurerm_app_service_slot` task the following slot on the same app service was created:
![image](https://user-images.githubusercontent.com/13313745/185196494-507015da-cd1b-462b-84c3-36f0bdf69724.png)

As you can see they have different App Service Plan's associated with them.

However, because `azurerm_windows_web_app_slot` has no ability to specify the service plan against the individual slot, we cannot update our terraform plans to use the new tasks as we cannot replicate this behaviour.

Does that help?

**Comment by NikRimington** [comment]
Also, in Azure, I believe you can change a slots app service plan using these controls here:
![image](https://user-images.githubusercontent.com/13313745/185197797-7dafcfab-48a8-4d25-8e2b-1c2fc3f09e19.png)

(Just so you can see where it is in the Azure portal) - I do not believe you can do this for production slots, only additional slots.

**Comment by jackofallops** [comment]
Closing as this functionality was added in #20184 and released in `v3.41.0`
